recipes/repl-builtin-modules/src/workflow.ts

```diff
@@ -0,0 +1,321 @@
+import { EOL } from "node:os";
+import { getNodeImportStatements } from "@nodejs/codemod-utils/ast-grep/import-statement";
+import { getNodeRequireCalls } from "@nodejs/codemod-utils/ast-grep/require-call";
+import { resolveBindingPath } from "@nodejs/codemod-utils/ast-grep/resolve-binding-path";
+import type { SgRoot, Edit, SgNode } from "@codemod.com/jssg-types/main";
+import type Js from "@codemod.com/jssg-types/langs/javascript";
+
+/**
+ * Transform function that converts deprecated repl.builtinModules and repl._builtinLibs
+ * to module.builtinModules API.
+ *
+ * See https://nodejs.org/api/deprecations.html#DEP0191
+ *
+ * Handles:
+ * 1. const repl = require('node:repl'); repl.builtinModules → const module = require('node:module'); module.builtinModules
+ * 2. const { builtinModules } = require('node:repl'); → const { builtinModules } = require('node:module');
+ * 3. const { builtinModules, foo } = require('node:repl'); → const { foo } = require('node:repl'); const { builtinModules } = require('node:module');
+ * 4. import { builtinModules } from 'node:repl'; → import { builtinModules } from 'node:module';
+ * 5. import { builtinModules, foo } from 'node:repl'; → import { foo } from 'node:repl'; import { builtinModules } from 'node:module';
+ * 6. const repl = require('node:repl'); repl._builtinLibs → const module = require('node:module'); module.builtinModules
+ * 7. const { _builtinLibs } = require('node:repl'); → const { builtinModules } = require('node:module');
+ * 8. import { _builtinLibs } from 'node:repl'; → import { builtinModules } from 'node:module';
+ */
+export default function transform(root: SgRoot<Js>): string | null {
+    const rootNode = root.root();
+    const edits: Edit[] = [];
+
+    // Helper functions
+    const getNewModuleSpecifier = (currentModule: string): string =>
+        currentModule.includes("node:") ? "'node:module'" : "'module'";
+
+    const containsBuiltinProperties = (text: string): boolean =>
+        text.includes("builtinModules") || text.includes("_builtinLibs");
```

Comment:
These can go outside of the transform function, to prevent recreating them on any transformation